internal/server/config.go

```diff
@@ -332,6 +360,18 @@ func (config *Config) setProperty(name, value string, fromLoad bool) error {
 		} else {
 			config._replicaPriority = replicaPriority
 		}
+	case AnnounceIP:
+		config._announceIP = value
+	case AnnouncePort:
+		if value == "" {
+			config._announcePort = 0
+		} else {
+			announcePort, err := strconv.ParseUint(value, 10, 64)
+			if err != nil {
+				invalid = true
+			}
+			config._announcePort = int64(announcePort)
```

Comment:
Should this line be inside of an else statement?

```go
if err != nil {
    invalid = true
} else {
    config._announcePort = int64(announcePort)
}
```

Otherwise the announce port will be updated to zero even when there's an invalid input.

Reply:
You are correct, fixing.